client/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
+import { AUTH_TOKEN, USER_ID } from './constants';
+
+
+@Injectable()
+export class AuthService {
+  private userId: string = null;
+  private _isAuthenticated = new BehaviorSubject(false);
```

Comment:
that is more common not to use '_' before private vars in TypeScript
I think that's better to discuss with team